File: Parte2/ReservaHotelFront/src/app/models/service.ts
import { Food } from "./food";

export interface ServiceItem{
    foodId: number;
    food?: Food;
    qtd: number;
}

export interface Service {
    id:number;
    userId:string;
    price:number;
    serviceItems: ServiceItem[];
    deliveryTime: string;
    status: string;
}

export interface ServiceCreate {
    userId: string;
    price: number;
    serviceItems: ServiceItem[];
    deliveryTime: string;
    status: string;
}

File: Parte2/ReservaHotelBack/ReservaHotel/Entities/Service.cs
namespace ReservaHotel.Entities
{
    public class Service
    {
        public int Id { get; set; }
        public float Price { get; set; }
        public DateTime DeliveryTime { get; set; }

        public User User { get; }
        public Guid UserId { get; set; }

        public ICollection<ServiceItem> ServiceItems { get; set; } = [];
    }
}
